### PR TITLE
Add LIP "Define state and state transitions of Random module"

### PR DESCRIPTION
**Pull request by IkerAlus**

### CODE REVIEW

File: proposals/lip-define_state_and_state_transitions_of_random_module.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+The entry in the random substore is defined as follows:
+
+* The store prefix is set to `STORE_PREFIX_RANDOM`.
+* The store key is set to empty bytes.
+* The store value is the serialization of an object following the JSON schema `seedRevealSchema` presented below.
+
+##### Seed Reveal Schema
+
+```
```
Comment (ricott1) [inline proposals/lip-define_state_and_state_transitions_of_random_module.md:74]:
```suggestion
```java
```

File: proposals/lip-define_state_and_state_transitions_of_random_module.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+* `generatorAddress`: A 20-bytes value with the address of the generator of a certain block.
+* `seedReveal`: A 16-bytes value with the seed revealed by the generator of a certain block.
+
+##### Returns
+
+This function returns `true` if `seedReveal` is valid input for the random seed computation, otherwise, `false`. 
+
+##### Execution
+
+```
```
Comment (ricott1) [inline proposals/lip-define_state_and_state_transitions_of_random_module.md:165]:
```suggestion
```python
```

File: proposals/lip-define_state_and_state_transitions_of_random_module.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+##### Returns
+
+This function returns `true` if `seedReveal` was correctly revealed, otherwise, `false`. 
+
+##### Execution
+
+It is specified as:
+
+```
```
Comment (ricott1) [inline proposals/lip-define_state_and_state_transitions_of_random_module.md:197]:
```suggestion
```python
```

File: proposals/lip-define_state_and_state_transitions_of_random_module.md [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+
+##### Returns
+
+`randomSeed`: A 32-bytes value representing the random seed. 
+
+##### Execution
+
+It is specified as:
+
+```
```
Comment (ricott1) [inline proposals/lip-define_state_and_state_transitions_of_random_module.md:225]:
```suggestion
```python
```

**Review by ricott1 — APPROVED** [review]
minor typos

**Review by MaximeGagnebin — APPROVED** [review]

**Review by ishantiw — APPROVED** [review]

**Review by MaximeGagnebin — APPROVED** [review]
Looks good!

**Review by ishantiw — APPROVED** [review]

**Review by karmacoma — APPROVED** [review]

**Pull request merged**